packages/ember-glimmer/lib/syntax/mount.js

```diff
@@ -113,25 +121,35 @@ class MountManager extends AbstractManager {
 
     engine.boot();
 
-    return engine;
+    return { engine, args };
   }
 
-  layoutFor(definition, engine, env) {
+  layoutFor(definition, { engine }, env) {
     let template = engine.lookup(`template:application`);
     return env.getCompiledBlock(OutletLayoutCompiler, template);
   }
 
-  getSelf(engine) {
+  getSelf(bucket) {
+    let { engine, args } = bucket;
+
     let applicationFactory = engine[FACTORY_FOR](`controller:application`);
-    let factory = applicationFactory || generateControllerFactory(engine, 'application');
-    return new RootReference(factory.create());
+    let controllerFactory = applicationFactory || generateControllerFactory(engine, 'application');
+    let controller = bucket.controller = controllerFactory.create();
```

Comment:
Ya, the bucket is a simple state bucket that we can use in these manager hooks.  A very similar thing happens in https://github.com/emberjs/ember.js/blob/master/packages/ember-glimmer/lib/syntax/curly-component.js (around L330).